wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/validator/DevfileSchemaValidator.java

```diff
@@ -75,6 +81,19 @@ private void validate(JsonNode contentNode) throws DevfileFormatException {
     try {
       List<Problem> validationErrors = new ArrayList<>();
       ProblemHandler handler = ProblemHandler.collectingTo(validationErrors);
+      if (!contentNode.hasNonNull("apiVersion")) {
+        throw new DevfileFormatException(
+            "Devfile schema validation failed. Error: The object must have a property whose name is \"apiVersion\".");
+      }
+      String apiVersion = contentNode.get("apiVersion").asText();
+
+      if (!schemasByVersion.containsKey(apiVersion)) {
+        throw new DevfileFormatException(
+            String.format(
+                "Version '%s' of the devfile is not supported. Supported versions are '%s'.",
+                apiVersion, SUPPORTED_VERSIONS));
```

Comment:
`SUPPORTED_VERSIONS` is a list. I would personally not wrap it in single quotes in the string.